tests/test_visualization.py

```diff
@@ -541,6 +541,29 @@ def test_plot_circles(self):
             plot_circles(target_position, target_radius, target_color, (-2, 2, -2, 2, -2, 2), ax=ax)
             savefig(write_dir, filename)
 
+    def test_plot_3D_targets(self):
+        from mpl_toolkits.mplot3d import Axes3D
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection='3d')
+
+        target_locations = np.array([(0, 0, 0), (2, 2, 2), (-3, -3, -3)])
+        colors = ['red', 'blue', 'green']
+        plot_targets_3D(target_locations, colors, target_radius=1.5, alpha=0.7, ax=ax)
+
+        filename = 'plot_3D_targets.png'
+        savefig(write_dir, filename)
```

Comment:
need to save this one to docs_dir instead of write_dir otherwise it won't get saved in the repo